src/NuGet.Clients/NuGet.VisualStudio.Client/.vsixinclude

```diff
@@ -34,5 +34,8 @@ NuGet.Versioning.dll
 
 # 3rd party dlls to keep
 Lucene.Net.dll
+Microsoft.Extensions.FileProviders.Abstractions.dll
+Microsoft.Extensions.FileSystemGlobbing.dll
+Microsoft.Extensions.Primitives.dll
```

Comment:
In nuget.exe's ilmerge config, there's also System.Memory, System.Buffers, and others.  These assemblies are already available in VS's PrivateAssemblies directory (and devenv.exe.config) when in VS. However, is there risk of any msbuild.exe command line scenario breaking?  Or does msbuild.exe also add VS's PrivateAssemblies path to the assembly load path?

The only file globbing codepaths I can think of are push and pack. Push doesn't exist in msbuild.exe. However, if a customer has a legacy csproj, and they use `-t:pack`, but they specify their own nuspec, which has file globbing, I think this is a codepath that is at risk.  Do we have any tests for such an edge case?

Reply:
I'll do an experimental insertion, install it, and play around with some scenarios like this.

Reply:
An insertion is really nothing more than installing NuGet.Tools.vsix. You can test, without a fake insertion, just by downloading the vsix from the build artifacts, and installing that on the VM. You just need to run the vspreinstall script to use the fake signed, PR build.

Reply:
I must be doing something wrong.  I did the following:

1. Install latest internal preview
2. Verified NuGet Package Manager works
3. Run vspreinstall script
4. Download NuGet.Tools.vsix from my build
5. Install NuGet.Tools.vsix

Now I get an error in ActivityLog and NuGet package manager won't load

```
SetSite failed for package [NuGetPackage]Source: 'System.ComponentModel.Composition' Description: The composition produced multiple composition errors, with 11 root causes. The root causes are provided below. Review the CompositionException.Errors property for more detailed information.
1) No exports were found that match the constraint: ContractName NuGet.VisualStudio.IConsoleStatus RequiredTypeIdentity NuGet.VisualStudio.IConsoleStatus
    Resulting in: Cannot set import 'NuGetVSExtension.NuGetPackage.ConsoleStatus (ContractName="NuGet.VisualStudio.IConsoleStatus")' on part 'NuGetVSExtension.NuGetPackage'.
    Element: NuGetVSExtension.NuGetPackage.ConsoleStatus (ContractName="NuGet.VisualStudio.IConsoleStatus") --> NuGetVSExtension.NuGetPackage
2) No exports were found that match the constraint: ContractName NuGet.PackageManagement.IDeleteOnRestartManager RequiredTypeIdentity NuGet.PackageManagement.IDeleteOnRestartManager
    Resulting in: Cannot set import 'NuGetVSExtension.NuGetPackage.DeleteOnRestartManager (ContractName="NuGet.PackageManagement.IDeleteOnRestartManager")' on part 'NuGetVSExtension.NuGetPackage'.
    Element: NuGetVSExtension.NuGetPackage.DeleteOnRestartManager (ContractName="NuGet.PackageManagement.IDeleteOnRestartManager") --> NuGetVSExtension.NuGetPackage
3) No exports were found that match the constraint: ContractName NuGet.PackageManagement.VisualStudio.INuGetUILogger RequiredTypeIdentity NuGet.PackageManagement.VisualStudio.INuGetUILogger
    Resulting in: Cannot set import 'NuGetVSExtension.NuGetPackage.OutputConsoleLogger (ContractName="NuGet.PackageManagement.VisualStudio.INuGetUILogger")' on part 'NuGetVSExtension.NuGetPackage'.
    Element: NuGetVSExtension.NuGetPackage.OutputConsoleLogger (ContractName="NuGet.PackageManagement.VisualStudio.INuGetUILogger") --> NuGetVSExtension.NuGetPackage
4) No exports were found that match the constraint: ContractName NuGet.ProjectManagement.INuGetProjectContext RequiredTypeIdentity NuGet.ProjectManagement.INuGetProjectContext
    Resulting in: Cannot set import 'NuGetVSExtension.NuGetPackage.ProjectContext (ContractName="NuGet.ProjectManagement.INuGetProjectContext")' on part 'NuGetVSExtension.NuGetPackage'.
	Element: NuGetVSExtension.NuGetPackage.ProjectContext (ContractName="NuGet.ProjectManagement.INuGetProjectContext") --> NuGetVSExtension.NuGetPackage
5) No exports were found that match the constraint: ContractName NuGet.Configuration.ISettings RequiredTypeIdentity NuGet.Configuration.ISettings
    Resulting in: Cannot set import 'NuGetVSExtension.NuGetPackage.Settings (ContractName="NuGet.Configuration.ISettings")' on part 'NuGetVSExtension.NuGetPackage'.
	Element: NuGetVSExtension.NuGetPackage.Settings (ContractName="NuGet.Configuration.ISettings") --> NuGetVSExtension.NuGetPackage
6) No exports were found that match the constraint: ContractName NuGet.PackageManagement.VisualStudio.IVsSolutionManager RequiredTypeIdentity NuGet.PackageManagement.VisualStudio.IVsSolutionManager
    Resulting in: Cannot set import 'NuGetVSExtension.NuGetPackage.SolutionManager (ContractName="NuGet.PackageManagement.VisualStudio.IVsSolutionManager")' on part 'NuGetVSExtension.NuGetPackage'.
	Element: NuGetVSExtension.NuGetPackage.SolutionManager (ContractName="NuGet.PackageManagement.VisualStudio.IVsSolutionManager") --> NuGetVSExtension.NuGetPackage
7) No exports were found that match the constraint: ContractName NuGet.PackageManagement.UI.SolutionUserOptions RequiredTypeIdentity NuGet.PackageManagement.UI.SolutionUserOptions
    Resulting in: Cannot set import 'NuGetVSExtension.NuGetPackage.SolutionUserOptions (ContractName="NuGet.PackageManagement.UI.SolutionUserOptions")' on part 'NuGetVSExtension.NuGetPackage'.
	Element: NuGetVSExtension.NuGetPackage.SolutionUserOptions (ContractName="NuGet.PackageManagement.UI.SolutionUserOptions") --> NuGetVSExtension.NuGetPackage
8) No exports were found that match the constraint: ContractName NuGet.VisualStudio.IVsSourceControlTracker RequiredTypeIdentity NuGet.VisualStudio.IVsSourceControlTracker
	Resulting in: Cannot set import 'NuGetVSExtension.NuGetPackage.SourceControlTracker (ContractName="NuGet.VisualStudio.IVsSourceControlTracker")' on part 'NuGetVSExtension.NuGetPackage'.
	Element: NuGetVSExtension.NuGetPackage.SourceControlTracker (ContractName="NuGet.VisualStudio.IVsSourceControlTracker") --> NuGetVSExtension.NuGetPackage
9) No exports were found that match the constraint: ContractName NuGet.PackageManagement.UI.INuGetUIFactory RequiredTypeIdentity NuGet.PackageManagement.UI.INuGetUIFactory
	Resulting in: Cannot set import 'NuGetVSExtension.NuGetPackage.UIFactory (ContractName="NuGet.PackageManagement.UI.INuGetUIFactory")' on part 'NuGetVSExtension.NuGetPackage'.
	Element: NuGetVSExtension.NuGetPackage.UIFactory (ContractName="NuGet.PackageManagement.UI.INuGetUIFactory") --> NuGetVSExtension.NuGetPackage
10) No exports were found that match the constraint: ContractName NuGet.VisualStudio.Common.IServiceBrokerProvider RequiredTypeIdentity NuGet.VisualStudio.Common.IServiceBrokerProvider
	Resulting in: Cannot set import 'NuGetVSExtension.NuGetPackage.ServiceBrokerProvider (ContractName="NuGet.VisualStudio.Common.IServiceBrokerProvider")' on part 'NuGetVSExtension.NuGetPackage'.
	Element: NuGetVSExtension.NuGetPackage.ServiceBrokerProvider (ContractName="NuGet.VisualStudio.Common.IServiceBrokerProvider") --> NuGetVSExtension.NuGetPackage
11) No exports were found that match the constraint: ContractName NuGet.VisualStudio.INuGetExperimentationService RequiredTypeIdentity NuGet.VisualStudio.INuGetExperimentationService
	Resulting in: Cannot set import 'NuGetVSExtension.NuGetPackage.NuGetExperimentationService (ContractName="NuGet.VisualStudio.INuGetExperimentationService")' on part 'NuGetVSExtension.NuGetPackage'.
	Element: NuGetVSExtension.NuGetPackage.NuGetExperimentationService (ContractName="NuGet.VisualStudio.INuGetExperimentationService") --> NuGetVSExtension.NuGetPackage
```

End-to-end and Apex tests passed so the VSIX isn't completely hosed right?  I must be doing something stupid.

Reply:
Weird. Maybe force rebuild the mef cache? 
https://marketplace.visualstudio.com/items?itemName=MadsKristensen.ClearMEFComponentCache#:~:text=Using%20the%20extension&text=Clicking%20the%20Clear%20MEF%20Component,cause%20any%20unwanted%20side%20effects.